Caboodle/Clipboard/Clipboard.ios.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Threading.Tasks;
+using UIKit;
+
+namespace Microsoft.Caboodle
+{
+    public static partial class Clipboard
+    {
+        public static Task SetTextAsync(string text) =>
+            Platform.InvokeOnMainThread(() => UIPasteboard.General.String = text);
+
+        public static bool HasText => UIPasteboard.General.HasStrings;
```

Comment:
I just noticed that the test is failing:
https://jenkins.mono-project.com/job/Components-Caboodle-DeviceTests-PR-Mac/15/testReport/(root)/(empty)/DeviceTests_Clipboard_Tests_Set_Clipboard_Values_text____text___/

It appears iOS does not support this on background threads. Instead of trying to make this async and doing extra work, I think it might be easier just to remove this property and just return null from `GetTestAsync` like UWP. And do the same for Android.

Reply:
This is not correct, we should not invoke on main thread here, just set the String.

We should ONLY do an InvokeOnMainThread when we have events and they come from a background. Always assume they call the API from UIThread, else they need to do it.